test/multiverse/suites/rake/instrumentation_test.rb

```diff
@@ -0,0 +1,245 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../../../../lib/new_relic/agent/instrumentation/rake/instrumentation'
+
+class RakeInstrumentationTest < Minitest::Test
+  class TesterClass
+    include NewRelic::Agent::Instrumentation::Rake::Tracer
+
+    def name; 'Snake'; end
+    def timeout; 140.85; end
+  end
+
+  class ErrorClass < StandardError; end
+
+  def test_invoke_with_newrelic_tracing_happy_path
+    instance = TesterClass.new
+    instance_mock = MiniTest::Mock.new
+    canned_config = {'rake.connect_timeout': instance.timeout}
+    NewRelic::Agent::Instrumentation::Rake.stub :should_trace?, true, [instance.name] do
+      NewRelic::Agent.stub :config, canned_config do
```

Comment:
Have you considered using the with_config block here instead of stubbing the agent config? Using the with_config block is a pretty standard pattern we use in most of our testing. it'd be like:
```
with_config('rake.connect_timeout': instance.timeout) do 
  # ... things ...
end
```

Reply:
➕

Reply:
with_config implemented with [2ba0e0f](https://github.com/newrelic/newrelic-ruby-agent/pull/1463/commits/2ba0e0f63ac56eeecf818cb0fa7faff15cadc147)